### PR TITLE
Update GH app installation instructions

### PR DESCRIPTION
**Pull request by pat-s**
- Slimify
- Add required permissions (missing so far)
- More detailed instructions on app settings
- Align env var options to style used in k8s section (UL instead of headers)

### CODE REVIEW

File: docs/docs/30-administration/11-forges/20-github.md [hunk 1]
```diff
@@ -1,72 +1,73 @@
...
 
-Read the value for `WOODPECKER_GITHUB_SECRET` from the specified filepath
+- `WOODPECKER_GITHUB_CLIENT_FILE` - Read the value for `WOODPECKER_GITHUB_CLIENT` from the specified filepath (Default: empty)
 
-### `WOODPECKER_GITHUB_MERGE_REF`
-> Default: `true`
+- `WOODPECKER_GITHUB_SECRET` - Configures the GitHub OAuth client secret. This is used to authorize access. (Default: empty)
 
-TODO
+- `WOODPECKER_GITHUB_SECRET_FILE` - Read the value for `WOODPECKER_GITHUB_SECRET` from the specified filepath 9Default: empty)
```
Comment (lafriks) [inline docs/docs/30-administration/11-forges/20-github.md:69]:
```suggestion
- `WOODPECKER_GITHUB_SECRET_FILE` - Read the value for `WOODPECKER_GITHUB_SECRET` from the specified filepath (Default: empty)
```

**Review by lafriks — APPROVED** [review]

File: docs/docs/30-administration/11-forges/20-github.md [hunk 2]
```diff
@@ -1,72 +1,73 @@
...
 
-Register your application with GitHub to create your client id and secret. It is very important that the authorization callback URL matches your http(s) scheme and hostname exactly with `<scheme>://<host>/authorize` as the path.
+## App Settings
 
-Please use this screenshot for reference:
+- Name: An arbitrary name for your App
+- Homepage URL: The URL of your Woodpecker instance
+- Callback URL: `https://<your-woodpecker-instance>/authorize`
+- Leave "Request user authorization (OAuth) during installation" and "Enable Device Flow" unchecked
+- Leave "Webhook" and "Post Installation" fields empty
```
Comment (runephilosof-karnovgroup) [inline docs/docs/30-administration/11-forges/20-github.md:21]: It should also mention how to set `Expire user authorization tokens`.

Comment (runephilosof-karnovgroup) [inline docs/docs/30-administration/11-forges/20-github.md:21]: The Webhook url field is required, cannot be left empty.

Comment (runephilosof-karnovgroup) [reply docs/docs/30-administration/11-forges/20-github.md:21]: It is set by Woodpecker (as an Oauth app) here:
https://github.com/woodpecker-ci/woodpecker/blob/main/server/api/repo.go#L116

The access token is generated there (in the PostRepo code), so you won't be able to fill in the field when creating Woodpecker as a Github App.

**Pull request merged**